### PR TITLE
feat(button group): create button group component

### PR DESCRIPTION
**Pull request by marinasmah**
Create new component "Button Group", according to [Spec](https://jira-eng-sjc12.cisco.com/jira/browse/CAX-453) [Design](https://www.figma.com/file/wDbhcTS6fYP9wFyJ7VnE0r/Lumos-Components?node-id=16085%3A0)

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/[sandbox]/examples/button-group.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import "@/components/button-group/ButtonGroup";
```
Comment (yananym) [inline web-components/src/[sandbox]/examples/button-group.ts:1]: please add a disclaimer comment to each new file

File: web-components/src/components/button-group/ButtonGroup.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    }
+  }
+
+  get selectedBtnValue() {
+    return (this.slotted[this.active] as CustomHTMLButtonElement).value;
+  }
+
+  private notifySelectedChange() {
+    this.dispatchEvent(
+      new CustomEvent("change", {
```
Comment (yananym) [inline web-components/src/components/button-group/ButtonGroup.ts:72]: need to give it a component-specific name

File: web-components/src/[sandbox]/examples/button-group.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <h3>Icon only - 2 options</h3>
+    <md-button-group>
+      <button slot="button" type="button"><md-icon name="icon-text-table_16"></md-icon></button>
+      <button slot="button" type="button"><md-icon name="icon-analysis_16"></md-icon></button>
+    </md-button-group>
+  </div>
+  <div class="column">
+    <h3>Icon only - disabled</h3>
+    <md-button-group disabled>
+      <button slot="button" type="button"><md-icon name="icon-text-table_16"></md-icon></button>
```
Comment (yananym) [inline web-components/src/[sandbox]/examples/button-group.ts:16]: what will happen if i put test inside button content? how to make it screen-reader accessible? these recommendations need to be present here.

**Review by yananym — CHANGES_REQUESTED** [review]

File: web-components/src/[sandbox]/examples/button-group.ts (hunk 1 above)
Comment (marinasmah) [reply web-components/src/[sandbox]/examples/button-group.ts:9]: Done

File: web-components/src/components/button-group/ButtonGroup.ts (hunk 2 above)
Comment (marinasmah) [reply web-components/src/components/button-group/ButtonGroup.ts]: Done

File: web-components/src/[sandbox]/examples/button-group.ts (hunk 3 above)
Comment (marinasmah) [reply web-components/src/[sandbox]/examples/button-group.ts]: Fixed

**Review by futurethang — CHANGES_REQUESTED** [review]
Lokks really great @marinasmah ! Only thing I really noticed was that the Dark Mode seems to have the 'hover' bg color active when it should be the darker 'normal' color in Figma, I think that's #1C1C1C instead of #292929

File: web-components/src/components/button-group/ButtonGroup.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+  get selectedBtnValue() {
+    return (this.slotted[this.active] as CustomHTMLButtonElement).value;
+  }
+
+  private notifySelectedChange() {
+    this.dispatchEvent(
+      new CustomEvent("button-group-change", {
+        detail: {
+          selected: this.selectedBtnValue
```
Comment (elenagiere) [inline web-components/src/components/button-group/ButtonGroup.ts:74]: I tried console logging `this.selectedBtnValue` and it seems like it is never defined, when I change the selection of the button group.

**Comment by marinasmah** [comment]
> Lokks really great @marinasmah ! Only thing I really noticed was that the Dark Mode seems to have the 'hover' bg color active when it should be the darker 'normal' color in Figma, I think that's #1C1C1C instead of #292929

Thanks @futurethang! Fixed

File: web-components/src/components/button-group/ButtonGroup.ts (hunk 4 above)
Comment (marinasmah) [reply web-components/src/components/button-group/ButtonGroup.ts:74]: Thanks @elenagiere! Fixed

**Review by elenagiere — DISMISSED** [review]

File: icons/svg/Queue-contact_20.svg [hunk 5]
```diff
@@ -1,5 +0,0 @@
-<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (yananym) [inline icons/svg/Queue-contact_20.svg:1]: please make sure you are not removing any unrelated files

**Review by elenagiere — DISMISSED** [review]

**Review by yananym — APPROVED** [review]

**Review by elenagiere — APPROVED** [review]

**Review by futurethang — APPROVED** [review]
